stdlib/getopt.pyi

```diff
@@ -1,10 +1,15 @@
-from collections.abc import Iterable
+from collections.abc import Iterable, Sequence
+from typing import TypeVar
+
+_StrSequenceT = TypeVar("_StrSequenceT", bound=Sequence[str])
 
 __all__ = ["GetoptError", "error", "getopt", "gnu_getopt"]
 
-def getopt(args: list[str], shortopts: str, longopts: Iterable[str] | str = []) -> tuple[list[tuple[str, str]], list[str]]: ...
+def getopt(
+    args: _StrSequenceT, shortopts: str, longopts: Iterable[str] | str = []
```

Comment:
I don't think the TypeVar makes sense. It doesn't return the same type as the input, it returns the result of slicing the input. The Sequence class doesn't require that slicing a `Sequence` of a particular type gives you the same type back.

Concretely, this would produce wrong results if the input type is say `tuple[str, str, str]`; the returned value may be an empty tuple, which is no longer a `tuple[str, str, str]`.

I would recommend using `Sequence[str]` as both the input and output type, without trying to make it generic. I haven't looked at the other files touched in this PR but I assume the same thing would apply there.

Reply:
Good point about slicing, let me try something